### ISSUE TITLE
Project Status

### ISSUE THREAD

**Issue by HennadiiM** [AlansCodeLog/shortcuts-visualizer#1]
As for me, wery useful feature. At least because you can see which keys are available, which not. Great visualization and user interface.

**Comment by alanscodelog** [comment]
Hi, I haven't given up. I was busy and now I'm re-writing it in a private repo/s. Also I ran into problems handling the dragging with this one.

I should probably update the readme. The new version is going to be split up into smaller libraries, leaving as little as possible mixed in with the Vue/UI parts so people can write a thin UI on top with other frameworks if they want, or none at all and just use the library to handle their user's shortcuts behind the scenes.

I have finished the boolean parser needed to handle the shortcut contexts (which will be published soon) and I'm working on the shortcut manager library itself now. I have already written all the code to handle "creating" keys, keyboards, and shortcuts. I just need to integrate the boolean parser and write how chains are handled, then it's back to where I was. You can follow/watch my profile if you're interested.

**Comment by tomasforsman** [comment]
Any news on the status of this project? It's exactly what I've been looking for but I can't get it to build properly. Fixed problems with python version, visual studio version and then node gyp version and after that when it seems like a problem with npm version I kind of gave up. It looks like an awesome project though and would love for it to work.

**Comment by alanscodelog** [comment]
I'm not sure it even worked in the last commit. Sorry. Feel free to take inspiration from the code, but this version was a mess. I've added a note to the readme.

I'm still working on the re-write. I did not have much time for it lately, but I just finished what was keeping me busy so I'll be getting back to it soon.

**Comment by jamiethomaswhite** [comment]
I need this in my life so bleeding badly!

Please, please, please consider getting this version in a working condition, regardless of the state of the code. I personally don't even need fancy things like drag and drop. I just desperately need a customisable visual log of all my keys and the shortcuts/macros etc I have assigned to them. The list below the main UI is a great addition I didn't think about as well.
This just looks exactly like what I've wanted for so long. It's so minimal and svelte, it also appears to function very nicely in the gif.

If it helps, I'm a gamer/streamer and this app would be essential for people like me. You might even consider creating a plugin that embeds this in other apps (such as Open Broadcaster Software).

If the new rewrite comes to fruition, even better, but even the original app would serve. I've tried downloading and running using nw.exe, but it didn't work. Otherwise, I don't know how to build.
There's really only one alternative option I've found, but it requires rewriting the code to customise it.

**Comment by alanscodelog** [comment]
Sorry, I just haven't had the time. I'm still working on the rewrite but in tiny bits. Trust me, I want this to work the most out of everyone or I wouldn't have tried to build it. I use a custom left hand dominant key map for everything (photoshop, krita, blender, etc) so I can use a mouse/tablet on the other hand, and I hate not being able to "unify" them all.

Getting this to a workable state again would be harder at this point than just continuing with my rewrite.

The gif is deceiving in that it's showing off the parts that worked (which that they did is no guarantee they currently do). Like if I patch a boat with duck tape, it'll appear to work to a spectator, but sooner or later it's going to sink.

I just checked out the public branch again and ran it. All the css it terribly broken, and there's debug messages  on screen from when I was working on it, etc, but the last commit where drag still seems to be "working" is 97bf460 if you're really desperate to try it. But it probably has bugs that make it unusable.

**Comment by yuvalco** [comment]
hey this project sounds exactly like what I need, is the rewriting still taking place?

**Comment by alanscodelog** [comment]
Yes, I just cleaned and made public the main repo [shortcuts-manager](https://github.com/AlansCodeLog/shortcuts-manager) and two related packages, a [boolean parser](https://github.com/alanscodelog/expressit) (for parsing shortcut contexts) and a [shortcuts parser](https://github.com/alanscodelog/parsekey).

They are all unpublished still and a bit WIP, more so the manager than anything. 

I'm still working on the ergonomics of the it and how the libraries connect. Also unlike this attempt it's not a framework specific library. There is no ui/frontend. The idea is it should handle as much as possible (data, events, etc) without locking you in to using a certain UI. Though some starter css might be provided at some point.

I will likely be archiving this repo shortly.
